### PR TITLE
fix: remove scrollbar from stats

### LINKED ISSUE #1376: Unnecessary scrollbar in ui.stats
### Discussed in https://github.com/h2oai/wave/discussions/1368

<div type='discussions-op-text'>

<sup>Originally posted by **azim-b** April 20, 2022</sup>
broswer: Safari Version 15.2
Wave version 0.21.0

Issue: `ui.stats()` rendering a scroll bar. Example:

image

Example 2:

image

**Cannot** reproduce this in wave 0.20.0

**Can** partially reproduce this in wave v0.21.0 in Firefox.
By partially, I mean that the scroll bars are rendered and then immediately disappear.</div>

### Repro

```py
q.page['body'] = ui.form_card(box='body',items=[
        # without inset
        ui.separator('Without inset'),
        ui.stats(justify='center',items=[
            ui.stat(label='H2O AI Health', value='AI4Good', icon='HealthSolid', icon_color='Red'),
        ]),
        # with inset set
        ui.separator('With inset'),
        ui.stats(justify='center', inset=True, items=[
            ui.stat(label='H2O AI Health', value='AI4Good', icon='HealthSolid', icon_color='Red'),
        ])
    ])
```

### PR DESCRIPTION
**Pull request by aalencar**
Taking into account #795, stats will not overflow horizontally since the label is going to wrap when it's too long.

After removing `overflow-x: auto`:
image

@mturoci let me know if I'm missing something on this one since you added `overflow-x: auto`.

closes #1376

### CODE REVIEW

**Comment by mturoci** [comment]
I am not sure why/how does removing `overflowX` solve the issue? According to screenshots, the problematic scrollbar is vertical and overflowX deals with horizontal one.

**Comment by aalencar** [comment]
> I am not sure why/how does removing `overflowX` solve the issue? According to screenshots, the problematic scrollbar is vertical and overflowX deals with horizontal one.

[Check this answers on Stackoverflow](https://stackoverflow.com/questions/6421966/css-overflow-x-visible-and-overflow-y-hidden-causing-scrollbar-issue/6433475#6433475).

Another solution would be to leave `overflow-x: auto` and set `overflow-y: hidden` because the card grows vertically to fit the content and won't overflow.

**Comment by mturoci** [comment]
The overflow-x is needed for cases when there are more stat items:
image

> [Check this answers on Stackoverflow](https://stackoverflow.com/questions/6421966/css-overflow-x-visible-and-overflow-y-hidden-causing-scrollbar-issue/6433475#6433475).

I see, didn't know about that one (+1). Unfortunately, this fix is more of "ignoring" the overflow rather than fixing it. See the following section for a detailed explanation.

> Another solution would be to leave overflow-x: auto and set overflow-y: hidden because the card grows vertically to fit the content and won't overflow.

overflow-y: hidden is just a way of ignoring the real problem. The real problem is that some element is overflowing the container when it shouldn't. So let's identify the problematic element, see why it overflows, and fix it accordingly. That's how unnecessary scrollbar problems should be handled in general.

**Comment by aalencar** [comment]
Changing the height of the svg container (span) from 1em to 100% fixed the issue.

File: ui/src/stats.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ export const
         <div key={`${i}:${label}`} className={css.stat} style={statStyle}>
           {icon && (
             <div className={css.lhs} style={icon_color ? { color: cssVar(icon_color) } : undefined}>
-              <FontIcon className={css.icon} iconName={icon} />
+              <Icon className={css.icon} iconName={icon} styles={{ root: { '> span': { height: '100%'}}}}/>
```
Comment (mturoci) [inline ui/src/stats.tsx:100]: The font icon styling is not necessary anymore. Make sure the rendered icon is 45x45.
```suggestion
              <Icon iconName={icon} styles={{ root: { '> span': { height: '100%'}}}}/>
```

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @aalencar, needs minor code clean up.

File: ui/src/stats.tsx (hunk 1 above)
Comment (aalencar) [reply ui/src/stats.tsx]: Changed `font-size` -> `height` + `width`
The icon is not rendered if the `height` or `width` is not set for `<i />`

**Review by mturoci — APPROVED** [review]
Thanks @aalencar!

**Pull request merged**